### PR TITLE
Cleanup System.IO.*Exception tests

### PR DESCRIPTION
**Pull request by hughbe**
- Take advantage of xunit
- Refactor tests to be more readable

Contributes to #6286
Cherry picked from #5490

### CODE REVIEW

File: src/System.Runtime/tests/System/IO/FileLoadException.Interop.cs [hunk 1]
```diff
@@ -4,53 +4,45 @@
...
-                HResults.ERROR_DISK_CORRUPT,
-                HResults.ERROR_UNRECOGNIZED_VOLUME,
-                HResults.ERROR_DLL_INIT_FAILED,
-                HResults.FUSION_E_CODE_DOWNLOAD_DISABLED,
-                HResults.CORSEC_E_MISSING_STRONGNAME,
-                HResults.MSEE_E_ASSEMBLYLOADINPROGRESS,
-                HResults.ERROR_FILE_INVALID,
-            };
+        var exception = Marshal.GetExceptionForHR(hr) as FileLoadException;
+        var fileLoadException = exception as FileLoadException;
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/IO/FileLoadException.Interop.cs]: Why are we casting to the same type twice?

Comment (hughbe) [inline src/System.Runtime/tests/System/IO/FileLoadException.Interop.cs]: Not sure. fixed

**Comment by hughbe** [comment]
Test Innerloop CentOS7.1 Debug Build and Test

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**